client-app/ui-kit/components/organisms/pagination/vc-pagination.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div v-if="pages > 1" class="vc-pagination">
+    <div class="vc-pagination__container">
+      <div class="vc-pagination__pages">
+        <component
+          :is="item > 0 && page !== item ? 'button' : 'span'"
+          v-for="(item, index) in visiblePages"
+          :key="index"
+          :type="item > 0 && page !== item ? 'button' : null"
+          :class="[
+            'vc-pagination__page',
+            {
+              'vc-pagination__page--active': item && page === item,
+              'vc-pagination__page--ellipsis': item === 0,
+            },
+          ]"
+          @click="setPage(item)"
+        >
+          {{ item || "..." }}
+        </component>
+      </div>
+
+      <div class="vc-pagination__nav">
+        <VcButton
+          class="vc-pagination__button vc-pagination__button--prev"
+          color="secondary"
+          variant="outline"
+          size="sm"
+          :disabled="page === 1"
+          @click="setPage(page - 1)"
+        >
+          <VcIcon name="chevron-left" />
+          <span>Previous</span>
+        </VcButton>
+
+        <VcButton
+          class="vc-pagination__button vc-pagination__button--next"
+          color="secondary"
+          variant="outline"
+          size="sm"
+          :disabled="page === pages"
+          @click="setPage(page + 1)"
+        >
+          <span>Next</span>
+          <VcIcon name="chevron-right" />
+        </VcButton>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+
+interface IEmits {
+  (event: "update:page", page: number): void;
+}
+
+interface IProps {
+  page?: number;
+  pages?: number;
+  scrollTarget?: HTMLElement;
+  scrollOffset?: number;
+}
+
+const emit = defineEmits<IEmits>();
+
+const props = withDefaults(defineProps<IProps>(), {
+  page: 1,
+  pages: 0,
+  scrollOffset: 20,
+});
+
+const visiblePages = computed(() => {
+  if (props.pages <= 9) {
+    return Array.from({ length: props.pages }, (_, i) => i + 1);
+  }
+
+  const pages = [];
+
+  if (props.page <= 5) {
+    for (let i = 1; i <= 7; i++) {
+      pages.push(i);
+    }
+
+    pages.push(0);
+    pages.push(props.pages);
+  } else if (props.page >= props.pages - 4) {
+    pages.push(1);
+    pages.push(0);
+
+    for (let i = props.pages - 6; i <= props.pages; i++) {
+      pages.push(i);
+    }
+  } else {
+    pages.push(1);
+    pages.push(0);
+
+    for (let i = props.page - 2; i <= props.page + 2; i++) {
+      pages.push(i);
+    }
+
+    pages.push(0);
+    pages.push(props.pages);
+  }
+
+  return pages;
+});
+
+function scrollToTop() {
+  if (props.scrollTarget) {
+    const topPosition = props.scrollTarget.getBoundingClientRect().top + window.scrollY - props.scrollOffset;
+    window.scrollTo({ top: topPosition, behavior: "smooth" });
+  }
+}
+
+const setPage = (page: number) => {
+  if (page) {
```

Comment:
Unnecessary check or wrong type: if page is number check is redundand, if i can be null or undefined - type is wrong.